libs/shared/src/main/java/io/crate/common/collections/Lists.java

```diff
@@ -104,6 +105,21 @@ public static <T> List<T> concatUnique(List<? extends T> list1, Collection<? ext
         return result;
     }
 
+    public static Collection<?> flattenUnique(Collection<?> list) {
+        if (!(list instanceof List<?>)) {
+            throw new IllegalArgumentException("Cannot flatten unless it is a nested array");
+        }
+        LinkedHashSet<Object> result = new LinkedHashSet<>();
```

Comment:
Did you benchmark if the de-duplication is actually leading to better performance? (vs. the cost it adds to do the deduplication here?)

Reply:
I believe this is no longer needed because of https://github.com/crate/crate/pull/16503.

Reply:
That mitigates it somewhat, but I think it would still be interesting to know if it shows any benefit to do the deduplication.

Reply:
Please see updates on #16503 for benchmark results.